enumerables.rb

```diff
@@ -0,0 +1,122 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    conv = if is_a? Range
+             to_a
+           else
+             self
+           end
+    i = 0
+    while i < conv.length
+      yield conv[i]
+      i += 1
+    end
+    conv
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    conv = if is_a? Range
+             to_a
+           else
+             self
+           end
+    i = 0
+    while i < conv.length
+      yield conv[i], i
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    selection = []
+    my_each { |item| selection << item if yield(item) }
+    selection
+  end
+
+  def my_all?(arg = nil)
+    check = true
+    if args
+      my_each { |i| check = false unless arg === i } # rubocop:disable Style/CaseEquality
+    else
+      my_each { |element| check = false unless yield(element) }
+    end
+    check
+  end
```

Comment:
* [ ] should have working `my_all?` identical and returns the same thing as ruby's [`all?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-all-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_all.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
```ruby
array = [1,2,3,4,5,6,7,8,9]
words = ['saheed', 'suretrust', 'oladele']

true_block = proc { |num| num <= 100 }
false_block = proc { |num| num > 100 }

true_array = [1, true, 'hi', []]
false_array = [1, false, 'hi', []]

array.my_all?(&true_block) != array.all?(&true_block)
array.my_all?(&false_block) != array.all?(&false_block)

true_array.my_all? != true_array.all?
false_array.my_all? != false_array.all?

array.my_all?(Integer) != array.all?(Integer)
array.my_all?(3) != array.all?(3)
words.my_all?(/d/) != words.all?(/d/)
```